app/livechat/server/api/v1/room.js

```diff
@@ -179,3 +182,54 @@ API.v1.addRoute('livechat/room.forward', { authRequired: true }, {
 		API.v1.success(Meteor.runAsUser(this.userId, () => Meteor.call('livechat:transfer', this.bodyParams)));
 	},
 });
+
+API.v1.addRoute('livechat/room.visitor', { authRequired: true }, {
+	put() {
+		try {
+			check(this.bodyParams, {
+				rid: String,
+				oldVisitorId: String,
```

Comment:
Since you're requesting the `oldVisitorId` here, you need to check if it matches with the current room visitor before replacing the current room visitor.

Reply:
Added this change in the latest commit